buildSrc/src/main/kotlin/Publishing.kt

```diff
@@ -40,12 +40,8 @@ fun MavenPom.configureMavenCentralMetadata(project: Project) {
 }
 
 fun mavenRepositoryUri(): URI {
-    // TODO -SNAPSHOT detection can be made here as well
     val repositoryId: String? = System.getenv("libs.repository.id")
     return if (repositoryId == null) {
-        // Using implicitly created staging, for MPP it's likely to be a mistake because
-        // publication on TeamCity will create 3 independent staging repositories
-        System.err.println("Warning: using an implicitly created staging for coroutines")
```

Comment:
I don't understand the changes to this file. They don't affect anything, but still, could you explain them?

Reply:
Our library is multiplatform and thus should be published from multiple machines (one per supported K/N architecture).

When we're configuring publication, we have to specify the repository we're going to publish to.
The default recommended/implicitly used address is `/staging/deploy/maven2/`. When it's used, a new staging repository is created by the Maven plugin. Essentially it means that, when we're publishing from multiple agents, we'll create multiple staging repositories (=> multiple releases with the same version and partial artifacts set).
To avoid that, we preconfigure our builds with a pre-created staging repository and pass its id via `libs.repository.id` for publishing tasks.

We do not do that for regular builds, but Gradle still configures publishing tasks, so this warning has been printed as many times as we got modules on each Gradle configuration, obfuscating our build output